### ISSUE TITLE
[Proposal] : no-useless-else

### ISSUE THREAD

**Issue by anikethsaha** [eslint/eslint#13067]
Relatable  : https://github.com/eslint/eslint/issues/8028, https://github.com/eslint/eslint/issues/11678

**Please describe what the rule should do:**
in `if-else` , if the body of `if` contains program control statements like `throw` , `break`, `continue`, `return` . then the `else` for that should be reported as use-less

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[X] Enforces code style (layout)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
if(foo){
  throw new Error('bar')
}else{           // report here about useless else
   console.log('else-statement')
}
```

```js
if(foo)
  continue
else   // report here
  return 'bar'
```

**Why should this rule be included in ESLint (instead of a plugin)?**
It is a universal flow and it seems preferable to have it in eslint

**Are you willing to submit a pull request to implement this rule?**
Yes, I will give it a try.

**Comment by anikethsaha** [comment]
any update on this ?

**Comment by mdjermanovic** [comment]
I support this :+1: 

As a new rule, it needs two more :+1: and a champion to be [accepted](https://eslint.org/docs/maintainer-guide/issues#accepting-issues).
